TACC/Core-CMS-Custom
# Task/WP-437-Exception-Form-React

## Overview

Create dynamic Exception Form React component 

## Related
- [WP-437](https://tacc-main.atlassian.net/browse/WP-437)

## Changes

- adds CDL hook and cdl class, url to common_api apps
- Creates exception form id to reference React component
- Updates template to only reference exception form id
- Adds ExceptionForm React component

## Testing

1. Go to [http://localhost:8000/submissions/exception/](http://localhost:8000/submissions/exception/)
2. Check that the dynamic fields (Field Code, Required Threshold Percentage, and Requested Threshold Percentage help text) display correctly with no console errors.
3. Review code noting the TODOs in the notes. PR was getting huge and wanted to break this down a bit.

## UI

<img width="1422" alt="Screenshot 2024-10-11 at 3 39 16 PM" src="https://github.com/user-attachments/assets/4b96d264-d8c5-4d26-9b55-6f1910633bfe">

## Notes

TODO:

- [ ] - Button actions
- [ ] - validating inputs
- [ ] - submitting form
- [ ] - adding non dynamic fields
- [ ] - option for non threshold exception submissions
